homeassistant/components/sensor/geizhals.py

```diff
@@ -0,0 +1,150 @@
+"""
+Parse prices of a device from geizhals.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.geizhals/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['beautifulsoup4==4.6.0']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_name = 'name'
+CONF_friendly_name = 'friendly_name'
+CONF_product_id = 'product_id'
+CONF_protocol = 'protocol'
+CONF_domain = 'domain'
+CONF_regex = 'regex'
+
+ICON = 'mdi:coin'
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=120)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_name): cv.string,
+    vol.Required(CONF_friendly_name, default='Price'): cv.string,
+    vol.Required(CONF_product_id): cv.positive_int,
+    vol.Optional(CONF_protocol, default='https'): vol.In(['https', 'http']),
+    vol.Optional(CONF_domain, default='geizhals.de'): vol.In(
+        ['geizhals.at',
+         'geizhals.eu',
+         'geizhals.de',
+         'skinflint.co.uk',
+         'cenowarka.pl']),
+    vol.Optional(CONF_regex, default='\D\s(\d*)[\,|\.](\d*)'): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Geizwatch Sensor."""
```

Comment:
I made some chnages in da424e167c335545483dfe7440d88a7d30e3df74, but I'm not quite sure what you meant here ...